communications/models.py

```diff
@@ -1,3 +1,69 @@
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
 from django.db import models
 
-# Create your models here.
+User = get_user_model()
+
+
+class ChatRoom(models.Model):
+    """
+        ChatRoom represents a room where a startup and an investor communicate.
+
+        Attributes:
+            investor (ForeignKey): The investor involved in the conversation.
+            startup (ForeignKey): The startup involved in the conversation.
+            created_at (DateTimeField): Timestamp when the chat room was created.
+        """
+    investor = models.ForeignKey(User, on_delete=models.SET_NULL, related_name='investor_chatrooms')
+    startup = models.ForeignKey(User, on_delete=models.SET_NULL, related_name='startup_chatrooms')
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        unique_together = ('investor', 'startup')
+        ordering = ['created_at']
+
+    def clean(self):
+        """
+        Ensure that one user has role - Investor and the other user has role - Startup.
+        """
+
+        is_investor = self.investor.roles.filter(name="Investor").exists()
+        is_startup = self.startup.roles.filter(name="Startup").exists()
+
+        if not (is_investor and is_startup):
+            raise ValidationError("One user must be an Investor, "
+                                  "and the other user must be a Startup.")
```

Comment:
Checking roles in `clean()` can result in extra queries. If you're saving many `ChatRoom` objects in bulk, this may lead to performance degradation.
The current logic does not ensure that the same user is not assigned both roles (`Investor` and `Startup`), which could break the logic.